ivy/container/linear_algebra.py

```diff
@@ -134,3 +134,174 @@ def cholesky(
+        key_chains: Optional[Union[List[str], Dict[str, str]]] = None,
+        to_apply: bool = True,
+        prune_unapplied: bool = False,
+        map_sequences: bool = False,
+        *,
+        out: Optional[ivy.Container] = None,
+    ) -> ivy.Container:
+
+        """
+        ivy.Container instance method variant of ivy.matrix_rank.
+        This method returns the rank (i.e., number of non-zero singular values) of a matrix (or a stack of
+        matrices).
+
+        Parameters
+        ----------
+        self
+            input container having shape ``(..., M, N)`` and whose innermost two dimensions form
+            ``MxN`` matrices. Should have a floating-point data type.
+
+        rtol
+            relative tolerance for small singular values. Singular values approximately less
+            than or equal to ``rtol * largest_singular_value`` are set to zero. If a
+            ``float``, the value is equivalent to a zero-dimensional array having a
+            floating-point data type determined by :ref:`type-promotion` (as applied to
+            ``x``) and must be broadcast against each matrix. If an ``array``, must have a
+            floating-point data type and must be compatible with ``shape(x)[:-2]`` (see
+            :ref:`broadcasting`). If ``None``, the default value is ``max(M, N) * eps``,
+            where ``eps`` must be the machine epsilon associated with the floating-point
+            data type determined by :ref:`type-promotion` (as applied to ``x``).
+            Default: ``None``.
+        
+        key_chains
+            The key-chains to apply or not apply the method to. Default is None.
+        
+        to_apply
+            If True, the method will be applied to key_chains, otherwise key_chains
+            will be skipped. Default is True.
+        
+        prune_unapplied
+            Whether to prune key_chains for which the function was not applied.
+            Default is False.
+
+        map_sequences
+            Whether to also map method to sequences (lists, tuples). Default is False.
+
+        out
+            optional output array, for writing the result to. It must have a shape that the
+            inputs broadcast to.
+
+        Returns
+        -------
+        ret
+            a container containing the ranks. The returned array must have a floating-point
+            data type determined by :ref:`type-promotion` and must have shape ``(...)``
+            (i.e., must have a shape equal to ``shape(x)[:-2]``).
+            
+        Examples
+        --------
+        With :code: `ivy.Container` input:
+
+        >>> x = ivy.Container(a=ivy.array([[1., 0.], [0., 1.]]), \
+                                b=ivy.array([[1., 0.], [0., 0.]]))
+        >>> y = x.matrix_rank()
+        >>> print(y)
+        {
+            a: ivy.array(2.),
+            b: ivy.array(1.)
+        }
+        
+        """ 
+
+        return self.matrix_rank(
```

Comment:
This should be `self.static_matrix_rank`, which is currently causing the docstrings tests to fail